src/chains/ethereum/console.log/src/maybe-get-logs.ts

```diff
@@ -0,0 +1,72 @@
+import { BN } from "ethereumjs-util";
+import { WORD_SIZE } from "./handlers";
+import { signatureMap } from "./signatures";
+
+const CONSOLE_ADDRESS = new BN(
+  Buffer.from([
+    0x63, 0x6f, 0x6e, 0x73, 0x6f, 0x6c, 0x65, 0x2e, 0x6c, 0x6f, 0x67
+  ])
+);
+
+/**
+ * The purpose of this type is to help maintainers visualize the layout of
+ * the `stack`.
+ */
+type LogsStack = [
+  ...rest: BN[],
+  inLength: BN,
+  inOffset: BN,
+  toAddress: BN,
+  _: BN
+];
+
+/**
+ * Returns arguments passed to console.sol's `console.log` if the given
+ * `@ethereumjs/vm` step event's `opcode`, `stack`, and `memory` fields contain
+ * valid `console.log` data.
+ *
+ * The event's `opcode.name` field must be `"STATICCALL"` and the `toAddress` on
+ * the `stack` must be the `console.log` address:
+ * `0x000000000000000000636f6e736f6c652e6c6f67`.
+ *
+ * @param event - `@ethereumjs/vm` step event
+ * @returns
+ */
+export const maybeGetLogs = (event: {
+  opcode: { name: string };
+  memory: Buffer;
+  stack: BN[];
+}): ConsoleLogs | null => {
+  if (event.opcode.name !== "STATICCALL") return null;
+
+  // STATICCALL, which is the OPCODE that is used to initiate a console.log, has
+  // 6 params, but we only care about these 3:
+  const stack = event.stack as LogsStack;
+  const [memoryLength, memoryOffset, toAddress] = stack.slice(-4, -1);
+
+  // only if the toAddress is our console address we should try parsing
+  if (!toAddress.eq(CONSOLE_ADDRESS)) return null;
+
+  // STATICCALL allows for passing in invalid pointers and lengths so we need to
+  // guard against failures with a try/catch
+  try {
+    const memoryStart = memoryOffset.toNumber();
+    const memoryEnd = memoryStart + memoryLength.toNumber();
+    const memory: Buffer = event.memory.subarray(memoryStart, memoryEnd);
+    const method = memory.readUInt32BE(0); // 4 bytes wide
+    const handlers = signatureMap.get(method);
+    if (!handlers) return null;
+
+    // we skip the first 4 bytes, as that is our signature
+    const values = memory.subarray(4);
+    return handlers.map((handler, index) => {
+      const offset = index * WORD_SIZE;
+      return handler(values, offset);
+    });
+  } catch {
+    return null;
```

Comment:
Should we log something here? I'm thinking if users are trying to call console.log incorrectly and it's silently failing.

Reply:
The users of our console.sol can't call it incorrectly.

They could use assembly and do it wrong though. I feel like that's on them though.

Or maybe HH, foundry, vyper, etc extend their own console.sol and we don't understand those extensions yet, in which case this `return null` behaves like it does in stable Ganache (ignores it).

We also ignore the case where the signature is "missing", which is pretty much the same issue, but we check for its existence instead of letting it throw.

Anyway, I was pretty much thinking about invalid data as adversarial, not mistakes.

Reply:
Sounds good